### PR TITLE
CVX Hodl WIP

### PR DESCRIPTION
**Pull request by CryptJS13**
Finished first hodl strat. Will continue working on the rest of them, but please have a look at the implementation whenever you have the time.

### CODE REVIEW

File: contracts/strategies/convex/base/ConvexStrategy2TokenHodl.sol [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+
+  function _setHodlVault(address _address) public onlyGovernance {
+    setAddress(_HODLVAULT_SLOT, _address);
+  }
+
+  function hodlVault() public view returns (address) {
+    return getAddress(_HODLVAULT_SLOT);
+  }
+
+  function _setHodlRatio(uint256 _value) public onlyGovernance {
```
Comment (breadforthepeople) [inline contracts/strategies/convex/base/ConvexStrategy2TokenHodl.sol:431]: Let's use underscores only for internal/private methods.
If a method is public, let's remove the underscore prefixes.

File: contracts/strategies/convex/base/ConvexStrategy2TokenHodl.sol [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+    require(_lpt == underlying(), "Pool Info does not match underlying");
+    require(_depositArrayPosition < 2, "Deposit array position out of bounds");
+    require(_hodlRatio <= 10000, "Needs to be smaller than 10000");
+    _setDepositArrayPosition(_depositArrayPosition);
+    _setPoolId(_poolID);
+    _setDepositToken(_depositToken);
+    _setDepositReceipt(_depositReceipt);
+    _setCurveDeposit(_curveDeposit);
+    _setHodlVault(_hodlVault);
+    _setPotPool(_potPool);
```
Comment (breadforthepeople) [inline contracts/strategies/convex/base/ConvexStrategy2TokenHodl.sol:96]: Is the intention to use a PotPool of the corresponding vault?
If yes, we need to:
1. Add validation to make sure it's the right PotPool
2. Not all vaults use PotPool. Some still use NoMintRewardPool

Comment (CryptJS13) [reply contracts/strategies/convex/base/ConvexStrategy2TokenHodl.sol:96]: Right, I had not thought about the old `NoMintRewardPool` problem. I have seen your new versions go up where a part of the CVX belonging to PS is sent to an address to HODL. This is certainly the most practical solution. I was thinking we could expand on this, and I am not sure you want to do this fully, but what we could do is hodl all of the PS rewards, 30% of the CRV and 30% of the CVX (if there are additional rewards like LDO/BOR just notify them separately for 30% buyback). This CRV and CVX can be sent to the multisig and the multisig can stake those assets with Convex for rewards. CRV is currently getting 95% APR and CVX is getting 80% APR. We can then use these rewards to buyback FARM and reward PS. This allows us to be more synergetic with both Curve and Convex, and should earn good rewards for PS over time.

When launching new Curve vaults we can look at turning this around, as the `PotPools` give us more possibilities. We could put the strategy rewards for users into CRV and CVX staking vaults, while the PS part just buys back farm.

**Comment by breadforthepeople** [comment]
We ended up going with a simpler version (already pushed). Closing this.

We are still discussing other approaches, and may need ConvexStrategyCVX.sol . Wondering if you could separate ConvexStrategyCVX (and all the related logic + the test), and make a pull request for it?

**Pull request closed without merging**